generator/src/main/java/com/scottlogic/deg/generator/inputs/AtomicConstraintReaderLookup.java

```diff
@@ -30,14 +30,14 @@
                 (dto, fields, rules) ->
                     new FormatConstraint(
                         fields.getByName(dto.field),
-                        (String) throwIfValueInvalid(dto.value),
+                        throwIfValueInvalid(dto, String.class),
```

Comment:
`throwIfValueInvalid` is now fully generic, and will return a String here (because of `String.class`).

The method will also assert that `dto.value` holds an instance of this type